File: ScheduleModule.DomainModels/Shift.cs
namespace ScheduleModule.DomainModels;

public class Shift
{
    public Guid ShiftId { get; set; }
    public Guid EmployeeId { get; set; }
    public DateOnly? Date { get; set; }
    public string FullName { get; set; }
    public string RoleName { get; set; }
    public TimeOnly StartHour { get; set; }
    public TimeOnly EndHour { get; set; }
}

File: ScheduleModuleClient/src/shared/models/shift.js
import { deserializeDate } from "../date-time-formatters.js";

export class Shift {
    constructor(urlParams) {
      this.shiftId = urlParams?.get('shiftId');
      this.employeeId =  urlParams?.get('employeeId');
      this.employeeName = urlParams?.get('employeeName');
      this.roleId;
      this.roleName = urlParams?.get('roleName');
      this.date = deserializeDate(urlParams?.get('date'));
      this.startHour = urlParams?.get('startHour');
      this.endHour = urlParams?.get('endHour');
    }
  }
  